bouncer.go

```diff
@@ -404,15 +404,21 @@ func handleBanServeHTTP(bouncer *Bouncer, rw http.ResponseWriter) {
 	}
 	rw.Header().Set("Content-Type", "text/html; charset=utf-8")
 	rw.WriteHeader(bouncer.remediationStatusCode)
-	_, err := fmt.Fprint(rw, bouncer.banTemplateString)
-	if err != nil {
-		bouncer.log.Error("handleBanServeHTTP could not write template to ResponseWriter")
+
+	// Don't write body content for HEAD requests
+	if method != http.MethodHead {
+		_, err := fmt.Fprint(rw, bouncer.banTemplateString)
+		if err != nil {
+			// use warn when https://github.com/maxlerebourg/crowdsec-bouncer-traefik-plugin/pull/276 is completed
+			bouncer.log.Info("handleBanServeHTTP could not write template to ResponseWriter: " + err.Error())
+		}
 	}
 }
 
 func handleRemediationServeHTTP(bouncer *Bouncer, remoteIP, remediation string, rw http.ResponseWriter, req *http.Request) {
 	bouncer.log.Debug(fmt.Sprintf("handleRemediationServeHTTP ip:%s remediation:%s", remoteIP, remediation))
-	if bouncer.captchaClient.Valid && remediation == cache.CaptchaValue {
+	// Only serve captcha for GET and POST requests, fallback to BAN for other methods. We need POST because the actual captcha post-processing is done in the POST request.
+	if bouncer.captchaClient.Valid && remediation == cache.CaptchaValue && (req.Method == http.MethodGet || req.Method == http.MethodPost) {
```

Comment:
Why you don't serve captcha page for PUT or DELETE ? Just do like for the ban page if not HEAD serve the captcha page ?

Reply:
In any case and to avoid further discussion and delays, I don't think this change has any relevant impact so I changed the code to only fallback to BAN for HEAD requests.